### PR TITLE
WIP: Sprawienie, żeby wszystkie testy działały z TypeScriptem

### PR DESCRIPTION
**Pull request by kuskoman**
Częściowo odnosi się do #320

### CODE REVIEW

File: frontend-project/src/utils/utils.test.ts [hunk 1]
```diff
@@ -1,16 +1,6 @@
 import { isUrl, getRouteAuthority } from './utils';
 
 describe('isUrl tests', () => {
-  it('should return false for invalid and corner case inputs', () => {
```
[the remark this answers is not in the document]
Comment (kuskoman) [reply frontend-project/src/utils/utils.test.ts:4]: @magul w typescripcie 3.8.3 nawet się nie skompilują. w 3.9 można uzyć ts expect errors, ale nie wiem po co mielibyśmy tego używać, jeśli język dba o poprawność typów, nie trzeba do tego pisać osobnych testów

File: frontend-project/jest.config.js [hunk 2]
```diff
@@ -1,9 +1,16 @@
...
     ANT_DESIGN_PRO_ONLY_DO_NOT_USE_IN_YOUR_PRODUCTION: false,
     localStorage: null,
   },
+  transform: {
+    '^.+\\.tsx?$': 'ts-jest',
+  },
+  moduleNameMapper: {
+    '^@/(.*)$': '<rootDir>/src/$1',
+  },
+  testMatch: ['**/?(*.)+(spec|test).[t]s'],
```
[the remark this answers is not in the document]
Comment (kuskoman) [reply frontend-project/jest.config.js:15]: racja, wziąłem regex z internetu

File: frontend-project/tsconfig.json [hunk 3]
```diff
@@ -8,6 +8,7 @@
     "baseUrl": ".",
     "jsx": "react",
     "allowSyntheticDefaultImports": true,
+    "esModuleInterop": true,
```
[the remark this answers is not in the document]
Comment (kuskoman) [reply frontend-project/tsconfig.json:11]: bez tego default imports nie chcą działać poprawnie, przynajmniej przy testach

[the remark this answers is not in the document]
Comment (kuskoman) [reply frontend-project/tsconfig.json:11]: ze strony typescripta:

> Emit `__importStar` and `__importDefault` helpers for runtime babel ecosystem compatibility and enable --allowSyntheticDefaultImports for typesystem compatibility.

`-allowSyntheticDefaultImports` też powinno działać zamiast tego

File: frontend-project/jest.config.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (kuskoman) [reply frontend-project/jest.config.js:15]: ale to chyba nie jest czysty jest, tylko z modyfikacjami od twórców frameworka i trzeba ustawiać tą właściwość (chyba, że to preset puppeteera to zmienia, albo plik z którego są testy uruchamiane)
w każdym razie bez tej właściwości nie działa tak jak powinno

**Comment by kuskoman** [comment]
Co do błędu w dockerze to libek w kontenerze po prostu brakuje, trzeba by dodać do dockerfile i powinno banglać

**Comment by kuskoman** [comment]
> Dodatkowo (i to jest w sumie pytanie, które próbuję zadać w #320) musimy się zastanowić, czy Chrome Headless wepchnąć do dockera, co chcielibyśmy testować w tych testach E2E i jakie to będzie doświadczenie dla programistów. Ja osobiście sie obawiam, że nie mamy zasobów, żeby te testy pisać/uruchamiać/utrzymywać i boję się, że albo one zarosną, albo będzie to dosyć frustrujące doświadczenie.

nawet wtedy bym proponował zmergować jak to skończę i ewentualnie potem usunąć, będzie łatwiej szukać w historii jakbyśmy chcieli do tego wrócić

File: frontend-project/src/utils/utils.test.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (kuskoman) [reply frontend-project/src/utils/utils.test.ts:4]: przykład:
```
 ● Test suite failed to run

    src/utils/utils.test.ts:5:18 - error TS2345: Argument of type 'undefined[]' is not assignable to parameter of type 'string'.

    5     expect(isUrl([])).toBeFalsy();
```

**Comment by kuskoman** [comment]
zamykam w związku z zamknięciem #320
